lib/src/chat/repository/message_repository.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:async';
+
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/material.dart';
+import 'package:reazzon/src/chat/message_bloc/message_entity.dart';
+
+abstract class MessageRepository {
+  String _userId;
+  String _loggedUserId;
+
+  String get userId => _userId;
+  String get loggedUserId => _loggedUserId;
+
+  Stream<List<MessageEntity>> getMessages();
+  Stream<bool> sendMessage(MessageEntity message);
+
+  static String generateGroupId(String from, String to) {
+    return (from.compareTo(to) > 0) ? '$from-$to' : '$to-$from';
+  }
+}
+
+class FireBaseMessageRepository extends MessageRepository {
```

Comment:
Again here the whole FirebaseMessageRepository needs to be made into async

Reply:
Sure.